### PR TITLE
DEV-378 Automate feddocs collections update

### PR DESCRIPTION
**Pull request by moseshll**
- Remove support for updating collections with headless browser
- Add support for talking to Collection Builder batch_collection.pl directly
- Remove many to most files in data/ to reduce clutter

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,22 +15,35 @@ The following collections are also owned by that user, but are static or manuall
 - [U.S. Census of Population: 1950](https://babel.hathitrust.org/cgi/mb?a=listis;c=1986287266)
 - [United States Decennial Census, 1790-1930](https://babel.hathitrust.org/cgi/mb?a=listis;c=1155585587)
 
-## get_item_lists.sh ##
-This is typically run Quarterly and retrieves HT Item IDs for the appropriate collections. Dependencies include:
+## process_collections.rb ##
```
Comment (mwarin) [inline README.md]: OK this is very nitpicky of me, but in markdown you don't need the right hand side `#`s for section heads.

File: get_item_lists.sh [hunk 2]
```diff
@@ -1,55 +1,49 @@
...
 
 # BOIA: 271744772
 bundle exec ruby get_ht_ids_by_auth.rb /l1/govdocs/authority_lists/txt/boia.auths.txt\
-  | sort -T tmpsort | uniq > data/boia_ht_item_ids.$D.txt 
-# comm -2 -3 data/boia_ht_item_ids.$D.txt data/boia_current_ht_item_ids.$D.txt\
-#  > data/boia_new_ht_item_ids.$D.txt
-# comm -1 -3 data/boia_ht_item_ids.$D.txt data/boia_current_ht_item_ids.$D.txt\
-#  > data/boia_deleted_ht_item_ids.$D.txt
- 
+    | sort -T tmpsort | uniq > data/boia_ht_item_ids.txt || {
```
Comment (mwarin) [inline get_item_lists.sh]: I am tempted to suggest breaking out common bash functions into a bash library file.

```
...
tmp_sort(){
    sort -T tmpsort | uniq
}
...
```

File: get_new_and_deleted.sh [hunk 3]
```diff
@@ -1,45 +1,41 @@
-D=$(date +\%Y-\%m-\%d)
-
 # BOIA: 271744772
-comm -2 -3 data/boia_ht_item_ids.$D.txt data/boia_current_ht_item_ids.$D.txt\
-  > data/boia_new_ht_item_ids.$D.txt
-comm -1 -3 data/boia_ht_item_ids.$D.txt data/boia_current_ht_item_ids.$D.txt\
-  > data/boia_deleted_ht_item_ids.$D.txt
+comm -2 -3 data/boia_ht_item_ids.txt data/boia_current_ht_item_ids.txt\
```
Comment (mwarin) [inline get_new_and_deleted.sh]: This file looks like 7 calls to the same function to me, if you want to take it that far.

File: process_collections.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
```
Comment (mwarin) [inline process_collections.rb:1]: Usually when I write ruby script that ends up shelling out to start another ruby script, I take that to mean it should have been a shell script all along. That's the feeling I get here.

File: update_collections.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+COLLECTIONS = {"all" => 2062901859,
```
Comment (mwarin) [inline update_collections.rb]: I've seen this hardcoded hash twice, and that's enough to make me want to break it out.

File: update_collections.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+               "epa" => 659388570,
+               "crc" => 1491645774,
+               "trail" => 1100951433}.freeze
+
+def base_command
+  who = `whoami`.chomp
+  "HT_DEV= MB_SUPERUSER=#{who} /htapps/babel/mb/scripts/batch-collection.pl -o feddocs@hathitrust.org "
+end
+
+def apply_additions(name:, id:)
```
Comment (mwarin) [inline update_collections.rb:10]: `apply_additions` and `apply_deletions` are very same-y but probably not worth DRY-ing out.

**Review by mwarin — APPROVED** [review]
Some nitpicks and suggestions for DRY-ing out on a rainy day, but no reason to stop releasing.

**Review by mwarin — APPROVED** [review]
Some nitpicks and suggestions for DRY-ing out on a rainy day, but no reason to stop releasing.

**Review by mwarin — APPROVED** [review]
Some nitpicks and suggestions for DRY-ing out on a rainy day, but no reason to stop releasing.

File: get_new_and_deleted.sh (hunk 3 above)
Comment (moseshll) [reply get_new_and_deleted.sh]: Yup, folding this in to another one of the higher level scripts and DRYing it out into a loop.

File: process_collections.rb (hunk 4 above)
Comment (moseshll) [reply process_collections.rb:1]: I am hopeless at doing anything other than copy-pasting shell. So yeah. Anything I can put in a Ruby string and invoke with backticks, I'm there.

File: update_collections.rb (hunk 6 above)
Comment (moseshll) [reply update_collections.rb:10]: Agreed. Especially since we may comment out the deletions stuff.

**Pull request merged**